config_scheduler.go

```diff
@@ -10,10 +10,18 @@ import (
 
 // SchedulerConfig holds relevant settings for the scheduler process.
 type SchedulerConfig struct {
-	TaskFinder           string  `bson:"task_finder" json:"task_finder" yaml:"task_finder"`
-	HostAllocator        string  `bson:"host_allocator" json:"host_allocator" yaml:"host_allocator"`
-	FreeHostFraction     float64 `bson:"free_host_fraction" json:"free_host_fraction" yaml:"free_host_fraction"`
-	CacheDurationSeconds int     `bson:"cache_duration_seconds" json:"cache_duration_seconds" yaml:"cache_duration_seconds"`
+	TaskFinder                    string  `bson:"task_finder" json:"task_finder" yaml:"task_finder"`
+	HostAllocator                 string  `bson:"host_allocator" json:"host_allocator" yaml:"host_allocator"`
+	FreeHostFraction              float64 `bson:"free_host_fraction" json:"free_host_fraction" yaml:"free_host_fraction"`
+	CacheDurationSeconds          int     `bson:"cache_duration_seconds" json:"cache_duration_seconds" yaml:"cache_duration_seconds"`
+	PlannerVersion                string  `bson:"planner_version" json:"planner_version" mapstructure:"planner_version"`
+	TargetTimeSeconds             int     `bson:"target_time_seconds" json:"target_time_seconds" mapstructure:"target_time_seconds"`
+	AcceptableHostIdleTimeSeconds int     `bson:"acceptable_host_idle_time_seconds" json:"acceptable_host_idle_time_seconds" mapstructure:"acceptable_host_idle_time_seconds"`
+	GroupVersions                 bool    `bson:"group_versions" json:"group_versions" mapstructure:"group_versions"`
+	PatchZipperFactor             int     `bson:"patch_zipper_factor" json:"patch_zipper_factor" mapstructure:"patch_zipper_factor"`
+	Interleave                    bool    `bson:"interleave" json:"interleave" mapstructure:"interleave"`
+	MainlineFirst                 bool    `bson:"mainline_first" json:"mainline_first" mapstructure:"mainline_first"`
+	PatchFirst                    bool    `bson:"patch_first" json:"patch_first" mapstructure:"patch_first"`
```

Comment:
Do these 3 need to be separate options? If I'm understanding correctly only 1 of these 3 should ever be true, which means they should be the same setting with 3 valid options

Reply:
Resolved into a single field called `TaskOrdering`.